crawler/crawler-ftp/src/main/java/fr/pilato/elasticsearch/crawler/fs/crawler/ftp/FileAbstractorFTP.java

```diff
@@ -167,13 +171,19 @@ public boolean exists(String dir) {
         }
     }
 
+    private boolean commandListenerEnabled() {
+        LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+        Configuration config = ctx.getConfiguration();
+        LoggerConfig loggerConfig = config.getLoggerConfig("fr.pilato.elasticsearch.crawler.fs");
+        return loggerConfig.getLevel() == Level.DEBUG || loggerConfig.getLevel() == Level.TRACE;
+    }
+
     @Override
     public void open() throws IOException {
-        Server server = fsSettings.getServer();
-        logger.debug("Opening FTP connection to {}@{}", server.getUsername(), server.getHostname());
-
         ftp = new FTPClient();
-        ftp.addProtocolCommandListener(ftpListener);
+        if (commandListenerEnabled()) {
```

Comment:
What about:

```java
if (logger.isTraceEnabled() || logger.isTraceEnabled()) {
  ftp.addProtocolCommandListener(ftpListener);
}
```

And remove the `commandListenerEnabled ()` method?

Reply:
LGTM!

Reply:
Could you push this change?

Reply:
Could you update the PR with this change?

Reply:
It's updated :)